mesonbuild/interpreter.py

```diff
@@ -2700,7 +2700,11 @@ def add_languages(self, args, required):
             else:
                 try:
                     (comp, cross_comp) = self.environment.detect_compilers(lang, need_cross_compiler)
-                    self.environment.check_compilers(lang, comp, cross_comp)
+                    if comp is None:
+                        raise EnvironmentException('Tried to use unknown language "%s".' % lang)
```

Comment:
These two are not really `EnvironmentException`s any more. They should be `InvalidArguments` like all the other argument failures in `interpreter.py`.

Reply:
Oh good point! It makes more sense that way and I think was before.